studio/components/interfaces/Settings/Logs/PreviewFilterPanel.tsx

```diff
@@ -97,9 +110,17 @@ const PreviewFilterPanel: FC<Props> = ({
       Refresh
     </Button>
   )
-  const handleSearch = (partial: Partial<Parameters<LogSearchCallback>[0]>) =>
-    onSearch({ query: search, to: partial?.to || null, from: partial?.from || null })
+  const handleSearch = (partial: Partial<Parameters<LogSearchCallback>[0]>) => {
+    const shouldShowUpgradePrompt = maybeShowUpgradePrompt(partial.from)
+
+    if (shouldShowUpgradePrompt && tier === 'FREE') {
+      setShowUpgradePrompt(!showUpgradePrompt)
+    }
 
+    if (!shouldShowUpgradePrompt) {
```

Comment:
should be combined with the above with an `else`, this would trigger each time the query range is more than 1 day.